botocore/handlers.py

```diff
@@ -180,6 +190,16 @@ def _looks_like_special_case_error(http_response):
     return False
 
 
+def _has_modeled_body(operation_model):
```

Comment:
To me it seems slightly odd for this function to live in `handlers.py`. I would consider moving this to `utils.py` or added as a property of the `OperationModel` class.